### ISSUE TITLE
Can't Add Public Methods Without Them Becoming Routes

### ISSUE THREAD

**Issue by eligundry** [teracyhq/flask-classful#41]
In previous versions (not sure of the range, because I don't know when this behavior was introduced) of Flask Classful, it was possible to have a public method in your view class and not have it automatically registered as a route. Take for example:

```python
from flask import request
from flask_classful import FlaskView

class BaseView(FlaskView):
    def copy_form_data(self):
         return request.form.copy()
```

In this example, `BaseView.copy_form_data` is registered as a route on all views that extend from `BaseView`. IMHO this isn't desired behavior, but whatevs.

So another issue related to this that I discovered (and maybe related to #29) is that I cannot make `BaseView.copy_form_data` into a static method because a `flask_classful.DecoratorCompatibilityError` is raised when it's encountered.

I have found that the only way around this restriction is to make that method private, which isn't super optimal because it makes it tricky to document with Sphinx and that it breaks existing code that depends on it.

### Discussion Points

1. Is it possible to whitelist methods that we don't want to be registered as routes? I would think that we would denote routes with the `route` decorator, but it appears that the ship has sailed on that design decision.
2. Should we automatically exclude static methods from being registered as routes?

**Comment by hoatle** [comment]
I see your point.

I guess we could:

1. Introduce a decorator or a class properties to exclude the method from auto route registered

2. We could just ignore any invalid method instead of raise exceptions, I'm not sure if this is good when users make mistake to their routes and we ignore them, it will make it difficult for users to know what's wrong. We should address this case carefully to make it robust to report errors if they're users' mistakes or just ignore if it's users' intentions.

What do you think about that?

**Comment by eligundry** [comment]
I like the thought process in point 1. Introducing a decorator makes a lot of sense for this. I also wouldn't be opposed to being able to add a class property that is a list of method names to ignore as routes.

**Comment by hjc1710** [comment]
:+1: for the decorator; and :+1: for the class attribute whitelist.

Definitely don't like just ignoring errors and making assumptions as that will be hard to debug.

**Comment by mattbillenstein** [comment]
Seems like there is some set of changes from this PR and perhaps #29 that should get merged -- what is the plan here?

**Comment by hoatle** [comment]
@mattbillenstein I guess we can merge #29 first and address this after that.

**Comment by hoatle** [comment]
for this issue, could anyone send a PR for this? I guess we can go with decorator and class attribute whitelist.

**Comment by hoatle** [comment]
Let's go with class attribute first, we could introduce decorator later on if required. This is the test for the changes:

```python
from flask import Flask

from flask_classful import FlaskView
from nose.tools import eq_


class NormalMethodsView(FlaskView):

    def copy_form_data(self):
        return "copy form data"


class ExcludedMethodsView(FlaskView):

    excluded_methods = ["copy_form_data"]

    def copy_form_data(self):
        return "copy form data"


app = Flask(__name__)
NormalMethodsView.register(app)
ExcludedMethodsView.register(app)

client = app.test_client()


def test_normal_methods_copy_form_data():
    resp = client.get("/normal-methods/copy_form_data/")
    eq_(b"copy form data", resp.data)


def test_excluded_methods_copy_form_data():
    resp = client.get("/excluded-methods/copy_form_data/")
    eq_(resp.status_code, 404)

```

**Comment by hoatle** [comment]
PR is created https://github.com/teracyhq/flask-classful/pull/57

@eligundry @hjc @mattbillenstein maybe you guys can help reviewing?

**Comment by hjc** [comment]
Absolutely! Apologies for not pitching in with the PR, I was on a long-form coding break when the request for functionality was issued and I missed it.

Happy to review your work now though.
